lsp-dart-utils.el

```diff
@@ -118,6 +118,14 @@ Order is important."
       (append list (list value))
     list))
 
+(defun lsp-dart-flutter-snap-install-p ()
+  ;; Detecting whether this is a Linux system with a Snap style install
+  (and (string= system-type "gnu/linux")
+       (let ((first-dir (-some-> (executable-find lsp-dart-flutter-executable) f-split cdr car)))
+	 (and first-dir
+	      (string= first-dir "snap")
+	      (file-exists-p "~/snap/flutter/common/flutter/bin/flutter")))))
```

Comment:
Sounds a little bit dangerous rely on this path, if snap changes anything we 'd need to update this